### PR TITLE
Add slides on defmt

### PR DESCRIPTION
**Pull request by jonathanpallant**
Adds a deck all about defmt

### CODE REVIEW

File: training-slides/src/defmt.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+defmt is the *Deferred Formatter*
+
+## Motivation
+
+* You have a microcontroller
+* You want to know what it is doing
+
+## Classical Approach
+
+* Set up a UART,
```
Comment (listochkin) [inline training-slides/src/defmt.md:14]: How widespread is the knowledge of what "UART" means? May be a note about it would be nice.

**Review by listochkin — APPROVED** [review]

**Comment by listochkin** [comment]
Such a nice deck overall! Best `defmt` explanation I've seen / heard.

File: training-slides/src/defmt.md (hunk 1 above)
Comment (jonathanpallant) [reply training-slides/src/defmt.md:14]: I think almost all embedded developers will have heard of a UART before.

**Comment by jonathanpallant** [comment]
I take no credit here. It's all from [the defmt book](https://defmt.ferrous-systems.com)!

**Comment by jonathanpallant** [comment]
Closed #122

**Pull request merged**